### PR TITLE
Issue #3209094 by Ressinel: Private group member doesn’t see group content on the search page

### PR DESCRIPTION
**Pull request by Ressinel**
## Problem
If I am a member of a private group the related content from that group should show up when I am browsing the search page. Currently, I can only see them if I go directly to the group.

## Solution
In the module `entity_access_by_field` add new records in `hook_node_access_records()`

## Issue tracker

- https://www.drupal.org/project/social/issues/3209094
- https://getopensocial.atlassian.net/browse/YANG-5309

## How to test
- [x] As a member of a private group try to search for some content which your group contain
- [ ] You will be able to see this content on the search page

## Screenshots
N/A

## Release notes
Fixed issue where private group member does not see group content on the search page.

## Change Record
N/A

## Translations
N/A

### CODE REVIEW

**Comment by Ressinel** [comment]
Hi @ronaldtebrake 

I see that only one failed test left for this PR but it doesn't relevant to my changes.
I made some investigation and see that for some other PRs the same test failed as well.
For example:
- https://github.com/goalgorilla/open_social/pull/2295/checks?check_run_id=2378984072
- https://github.com/goalgorilla/open_social/pull/2292/checks?check_run_id=2341930970
- https://github.com/goalgorilla/open_social/pull/2293/checks?check_run_id=2343411696

Do we need whether skip it and continue the review or do it after the test will be fixed?

Thanks!

**Review by ribel — APPROVED** [review]

File: modules/custom/entity_access_by_field/entity_access_by_field.install [hunk 1]
```diff
@@ -100,3 +100,10 @@ function _entity_access_by_field_get_permissions($role) {
   }
   return [];
 }
+
+/**
+ * Rebuild node access.
+ */
+function entity_access_by_field_update_8004() {
```
Comment (navneet0693) [inline modules/custom/entity_access_by_field/entity_access_by_field.install:107]: Please move it to post_update_hook_name.

**Review by navneet0693 — CHANGES_REQUESTED** [review]
We also need to add code for re-indexing data in search indexes.

File: modules/custom/entity_access_by_field/entity_access_by_field.install (hunk 1 above)
Comment (Ressinel) [reply modules/custom/entity_access_by_field/entity_access_by_field.install]: I have moved it.

**Comment by Ressinel** [comment]
Hi @navneet0693 
I have added code for re-indexing data in necessary search indexes.

**Comment by slowflyer** [comment]
Hi, I just read re-indexing and the question came up, if it will be compatible and neccessary when moving to search_api 8.1.19? https://github.com/goalgorilla/open_social/pull/2307

**Comment by navneet0693** [comment]
> Hi, I just read re-indexing and the question came up, if it will be compatible and neccessary when moving to search_api 8.1.19? #2307

@slowflyer This PR is aimed for OpenSocial version less than 10.2.x which will still have an older version of search_api.

But yes, we can also simply add it to releases notes that it is mandatory for re-indexing after this change. This will also be good from a performance point of view to keep this process separate from updates.

Re-indexing after any changes to indexes is generally a good idea, just to make sure everything is updated & changes have taken effect.

Also, we do need an update hook to update the existing website's search indexes once 10.2.x is released. So, we definitely need a follow-up of #2307. This will also make sure that the upgrade path is maintained.

Please let us know if you see any issues forthcoming?

**Comment by Kingdutch** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2295-twddtfelesvkmazw5f17x60tpzyqxepk.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60c73c79cde720d3e97e6447

**Comment by Ressinel** [comment]
Hi @navneet0693 
As you removed code for re-indexing then maybe this PR ready to merge?

**Comment by navneet0693** [comment]
@Ressinel Yes! Let's do it :-D

**Review by navneet0693 — APPROVED** [review]

**Comment by navneet0693** [comment]
🍒 - picked on 10.2.x and 10.1.x!

**Pull request merged**